app/controllers/ai_annotations_controller.rb

```diff
@@ -1,43 +1,71 @@
 class AiAnnotationsController < ApplicationController
+  before_action :authenticate_user!, except: [ :new ]
+
+  def index
+    @ai_annotations = AiAnnotation.where(user: current_user)
+  end
+
   def new
     @new_ai_annotation = AiAnnotation.new
     @history = AiAnnotation.order(created_at: :desc).limit(10)
-    @llm_api_keys = fetch_llm_api_keys
+    @llm_api_keys = LlmMetaServerResource.llm_api_keys current_user
   end
 
   def create
     text = ai_annotation_params[:text]
     prompt = ai_annotation_params[:prompt]
     @new_ai_annotation = AiAnnotation.prepare_with(text, prompt)
 
-    ai_annotation = @new_ai_annotation.annotate!
-    increment_token_usage(@new_ai_annotation.token_used)
+    @selected_api_key_uuid = params[:api_key_uuid]
+    @selected_model = params[:model]
+
+    token = current_user.id_token
+    ai_annotation = @new_ai_annotation.annotate! token, @selected_api_key_uuid, @selected_model
 
     redirect_to "/ai_annotations/#{ai_annotation.uuid}"
   rescue => e
     Rails.logger.error "Error: #{e.message}"
     flash.now[:alert] = "Unexpected error occurred while generating AI annotation."
+
+    # Set required variables in case of error
+    @history = user_signed_in? ? AiAnnotation.order(created_at: :desc).limit(10) : []
+
     render :new, status: :unprocessable_entity
   end
 
   def edit
-    @ai_annotation = AiAnnotation.find_by!(uuid: params[:uuid])
+    @ai_annotation = AiAnnotation.find_by(uuid: params[:uuid])
+    unless @ai_annotation
+      redirect_to root_path
+      return
+    end
+
     @history = AiAnnotation.order(created_at: :desc).limit(10)
-    @llm_api_keys = fetch_llm_api_keys
+    @llm_api_keys = LlmMetaServerResource.llm_api_keys current_user
   end
 
   def update
-    @ai_annotation = AiAnnotation.find_by(uuid: params[:id])
+    @ai_annotation = AiAnnotation.find_by(uuid: params[:uuid])
+    unless @ai_annotation
+      redirect_to root_path
+      return
+    end
```

Comment:
```suggestion
    return redirect_to root_path unless @ai_annotation
```

と、書くとRailsっぽくなります。
アーリリターンの一種です。